services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -212,9 +216,62 @@ def engine(self) -> Engine:
             async for row in conn.execute(query):
                 result_dict = dict(row.items())
                 log.debug("found project: %s", result_dict)
+                tags = await self._get_tags_by_study(study_id=result_dict['id'])
+                result_dict['tags'] = tags
                 projects_list.append(_convert_to_schema_names(result_dict))
         return projects_list
 
+    async def _get_study(self, user_id: str, project_uuid: str, exclude_foreign: List = []) -> Dict: # pylint: disable=dangerous-default-value
+        async with self.engine.acquire() as conn:
+            joint_table = user_to_projects.join(projects)
+            query = select([projects]).select_from(joint_table).where(
+                and_(projects.c.uuid == project_uuid,
+                     user_to_projects.c.user_id == user_id)
+            )
+            result = await conn.execute(query)
+            study_row = await result.first()
+
+            if not study_row:
+                raise ProjectNotFoundError(project_uuid)
+
+            study = dict(study_row.items())
+
+            if 'tags' not in exclude_foreign:
+                tags = await self._get_tags_by_study(study_id=study_row.id)
+                study['tags'] = tags
+
+            return study
+
+
+    async def add_tag(self, user_id: str, project_uuid: str, tag_id: int) -> Dict:
+        study = await self._get_study(user_id, project_uuid)
+        async with self.engine.acquire() as conn:
+            # pylint: disable=no-value-for-parameter
+            query = study_tags.insert().values(
+                study_id=study['id'],
+                tag_id=tag_id
+            )
+            async with conn.execute(query) as result:
+                if result.rowcount == 1:
+                    study['tags'].append(tag_id)
+                    return _convert_to_schema_names(study)
+                raise web.HTTPInternalServerError()
```

Comment:
This is the db API. We should not raise HTTP errors. Use ``projects_exceptions`` instead

Reply:
raising ``ProjectException`` (unexpected error)

Reply:
Please do not forget to raise non http exceptions

Reply:
https://github.com/ignapas/osparc-simcore/blob/is1224/labels/services/web/server/src/simcore_service_webserver/projects/projects_db.py#LC260
This should be changed already. I don't know why it shows like that... GitHub's interface is sometimes misleading